src/modules/accounting.py

```diff
@@ -0,0 +1,251 @@
+            beacon_validators_count,
+            beacon_validators_balance,
+            stacking_module_ids,
+            no_operators,
+            exited_validators,
+            wc_balance,
+            wei_to_reserve,
+            last_requests_id_to_finalize,
+            finalization_shares_amount,
+        ))
+
+        return transaction
+
+    def _get_beacon_validators_stats(self, slot: SlotNumber, block_hash: HexBytes) -> Tuple[int, int]:
+        # Should return balance in gwei!
+        lido_validators = get_lido_validators(self._w3, block_hash, self._beacon_chain_client, slot)
+        return len(lido_validators), sum(int(validator['validator']['balance']) for validator in lido_validators)
+
+    def _get_new_exited_validators(
+            self,
+            slot: SlotNumber,
+            block_hash: HexBytes,
+    ) -> Tuple[List[str], List[int], List[int]]:
+        validators = self._get_exited_validators(slot, block_hash)
+
+        exited_validators = defaultdict(int)
+
+        for validator in validators:
+            exited_validators[(validator['key']['module_id'], validator['key']['operator_index'])] += 1
+
+        stacking_module_id = []
+        node_operator_id = []
+        exited_validators_number = []
+
+        operators = get_lido_node_operators(self._w3, block_hash)
+        for operator in operators:
+            exited_validators_prev = operator['stoppedValidators']
+
+            exited_validators_current = exited_validators[(operator['module_id'], operator['index'])]
+
+            if exited_validators_prev != exited_validators_current:
+                stacking_module_id.append(operator['module_id'])
+                node_operator_id.append(operator['index'])
+                exited_validators_number.append(exited_validators_current)
+
+        return (
+            stacking_module_id,
+            node_operator_id,
+            exited_validators_number,
+        )
+
+    def _get_exited_validators(self, slot: SlotNumber, block_hash: HexBytes) -> List[MergedLidoValidator]:
+        lido_validators = get_lido_validators(self._w3, block_hash, self._beacon_chain_client, slot)
+
+        withdrawable_validators = filter(lambda validator: int(validator['validator']['validator']['exit_epoch']) <= int(slot / self.slots_per_epoch), lido_validators)
+
+        return list(withdrawable_validators)
+
+    def _get_wc_balance(self, block_hash: HexBytes) -> int:
+        return self._w3.eth.get_balance(
+            contracts.lido.functions.getELRewardsVault().call(block_identifier=block_hash),
+            block_identifier=block_hash,
+        )
+
+    def _get_wei_to_reserve(self, block_hash: HexBytes) -> int:
+        return get_withdrawal_requests_wei_amount(block_hash)
+
+    def _get_requests_finalization_report(self, block_hash: HexBytes) -> Tuple[List[int], List[int]]:
+        last_finalized_request = contracts.withdrawal_queue.functions.finalizedQueueLength().call(block_identifier=block_hash)
```

Comment:
finalizedQueueLength - didn't find references in smart-contracts.

Reply:
https://github.com/lidofinance/lido-dao/blob/feature/shapella-upgrade/contracts/0.8.9/WithdrawalQueue.sol#L220

All of the changes should be relevant to the `shapella-upgrade` branch.